packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-media-text.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * This file is part of the WooCommerce Email Editor package
+ *
+ * @package Automattic\WooCommerce\EmailEditor
+ */
+
+declare( strict_types = 1 );
+namespace Automattic\WooCommerce\EmailEditor\Integrations\Core\Renderer\Blocks;
+
+use Automattic\WooCommerce\EmailEditor\Engine\Renderer\ContentRenderer\Rendering_Context;
+use Automattic\WooCommerce\EmailEditor\Integrations\Utils\Table_Wrapper_Helper;
+use Automattic\WooCommerce\EmailEditor\Integrations\Utils\Styles_Helper;
+use Automattic\WooCommerce\EmailEditor\Integrations\Utils\Dom_Document_Helper;
+
+/**
+ * Media-text block renderer.
+ * This renderer handles core/media-text blocks with proper email-friendly HTML layout.
+ */
+class Media_Text extends Abstract_Block_Renderer {
+	/**
+	 * Renders the media-text block content using a direct table-based layout.
+	 *
+	 * @param string            $block_content Block content.
+	 * @param array             $parsed_block Parsed block.
+	 * @param Rendering_Context $rendering_context Rendering context.
+	 * @return string
+	 */
+	protected function render_content( string $block_content, array $parsed_block, Rendering_Context $rendering_context ): string {
+		$block_attrs  = $parsed_block['attrs'] ?? array();
+		$inner_blocks = $parsed_block['innerBlocks'] ?? array();
+
+		// Extract media content from innerHTML.
+		$media_content = $this->extract_media_from_html( $parsed_block['innerHTML'] ?? $block_content );
+
+		// Render all inner blocks content.
+		$text_content = '';
+		foreach ( $inner_blocks as $block ) {
+			$text_content .= render_block( $block );
+		}
+
+		// If we don't have both media and text content, return empty.
+		if ( empty( $media_content ) || empty( $text_content ) ) {
+			return '';
+		}
+
+		// Build the email-friendly layout.
+		return $this->build_email_layout( $media_content, $text_content, $block_attrs, $block_content, $rendering_context );
+	}
+
+	/**
+	 * Extract media content from the HTML block content.
+	 *
+	 * @param string $block_content Raw block content.
+	 * @return string Media HTML content or empty string if not found.
+	 */
+	private function extract_media_from_html( string $block_content ): string {
+		// Extract media content (preserving any link wrapper).
+		$media_content = '';
+		if ( preg_match( '/<figure[^>]*class="[^"]*\bwp-block-media-text__media\b[^"]*"[^>]*>.*?<\/figure>/s', $block_content, $matches ) ) {
```

Comment:
Would it make sense leave out the `<figure>` and extract only the inner content? It's [not well supported in all email clients](https://www.caniemail.com/search/?s=figure) and seems to be adding unexpected styles:

<img width="452" height="233" alt="ncaq7RDDlEi0QyJF" src="https://github.com/user-attachments/assets/063ee035-40b4-4d85-9e74-6c9b6dee7fa7" />